core/src/test/java/org/apache/calcite/sql/type/SqlTypeUtilTest.java

```diff
@@ -117,6 +120,45 @@
     SqlTypeCoercionRule.THREAD_PROVIDERS.set(defaultRules);
   }
 
+  @Test void testEqualAsCollectionSansNullability() {
+    final SqlTypeFactoryImpl factory =
+        new SqlTypeFactoryImpl(RelDataTypeSystem.DEFAULT);
+
```

Comment:
Try to reuse the `Fixture`.

Reply:
The basic type is replaced, but nested type in SqlTypeFixture does not meet the requirements(map type/nullability), should I add these to SqlTypeFixture? Or just leave them in this ut.

Reply:
I would prefer add the types to `SqlTypeFixture` and give it a good name.

Reply:
> I would prefer add the types to `SqlTypeFixture` and give it a good name.

Thanks for your suggestion, updated already.

Reply:
> I would prefer add the types to `SqlTypeFixture` and give it a good name.

Thanks for your suggestion, updated already.